great_expectations/datasource/new_datasource.py

```diff
@@ -411,11 +423,12 @@ class Datasource(BaseDatasource):
     def __init__(
         self,
         name: str,
-        execution_engine=None,
-        data_connectors=None,
+        execution_engine: Optional[dict] = None,
+        data_connectors: Optional[dict] = None,
```

Comment:
## Non-blocking

Same comment as elsewhere. If there is a predictable structure to these `dict` can we define a `TypeDict` for them.

Reply:
Same note!